### PR TITLE
Add Loans controller

### LINKED ISSUE #18: Create Loan controller and actions
Endpoints to implement for now:

 - [x] Create
 - [x] Index
- [x] Update
- [x] Destroy
- [ ] Add new endpoints to Insomnia
- [x] Implement test cases for endpoints

### PR DESCRIPTION
**Pull request by ernest-phillips**
This closes #18

### CODE REVIEW

File: spec/factories/creditors.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require 'faker'
+
+FactoryBot.define do
+  Faker::Company.name
```
Comment (gabrielgaraujo) [inline spec/factories/creditors.rb:4]: `Faker::Company.name` is a value. Therefore, you need to tell FactoryBot a value for "what"

```suggestion
  name { Faker::Company.name }
```

Same goes for `url` in this file and the attributes on `spec/factories/users.rb`

File: spec/requests/loans_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          creditor_id: @creditor.id,
+          user_id: @user.id
+        }
+      }
+    end
+
+    it 'creates a new loan' do
+      expect{ post('/loans', params: loan_params) }.to change { Loan.count }.by(1)
+    end
+  end
```
Comment (gabrielgaraujo) [inline spec/requests/loans_spec.rb:27]: Shall we also add test cases listing, updating and deleting a record?
While adding test cases for these, you may need to create a new factory `:loans`, refer to https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#associations to understand how to define associations on factory bot

File: spec/factories/loans.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'faker'
+
+FactoryBot.define do
+  factory :loan do
+    name { Faker::Company.buzzword }
```
Comment (ernest-phillips) [inline spec/factories/loans.rb:5]: @gabrielgaraujo I tried to use `@name` in the spec file, but I kept getting undefined method. I wanted to replace the hard coded text for name.

Comment (gabrielgaraujo) [reply spec/factories/loans.rb:5]: Everything looks great, lets check this issue you are facing together!

**Review by gabrielgaraujo — APPROVED** [review]

**Pull request merged**